java-ast/addon/src/main/java/org/jboss/windup/ast/java/ReferenceResolvingVisitor.java

```diff
@@ -1131,17 +991,20 @@ public boolean visit(MethodInvocation node)
         if (resolveTypeBinding != null)
         {
             resolutionStatus = ResolutionStatus.RESOLVED;
-            ITypeBinding[] arguments = resolveTypeBinding.getParameterTypes();
+            ITypeBinding[] argumentTypeBindings = resolveTypeBinding.getParameterTypes();
+            List<Expression> arguments = node.arguments();
 
-            for (ITypeBinding type : arguments)
+            int index = 0;
+            for (ITypeBinding type : argumentTypeBindings)
             {
+                Expression expression = arguments.get(index++);
```

Comment:
this is dangerous as it doesn't check the arguments size while iterating through argumentTypeBindings. This fails for some method invocations like https://docs.jboss.org/hibernate/search/4.3/api/org/hibernate/search/FullTextSession.html#createFullTextQuery(org.apache.lucene.search.Query,%20java.lang.Class...)
where are variable arguments.